component/src/main/java/org/wso2/extension/siddhi/gpl/execution/streamingml/util/CoreUtils.java

```diff
@@ -0,0 +1,184 @@
+            }
+        }
+        return featureVariableExpressionExecutors;
+    }
+
+
+    public static boolean isNumeric(Attribute.Type attributeType) {
+        return numericTypes.contains(attributeType);
+    }
+
+    public static boolean isLabelType(Attribute.Type attributeType) {
+        return labelTypes.contains(attributeType);
+    }
+
+
+    /**
+     * @param inputDefinition
+     * @param attributeExpressionExecutors
+     * @param classIndex
+     * @return
+     */
+    public static VariableExpressionExecutor extractAndValidateClassLabel
+    (AbstractDefinition inputDefinition, ExpressionExecutor[] attributeExpressionExecutors, int classIndex) {
+        VariableExpressionExecutor classLabelVariableExecutor;
+
+        if (attributeExpressionExecutors[classIndex] instanceof VariableExpressionExecutor) {
+            // other attributes should be numeric type.
+            String attributeName = ((VariableExpressionExecutor)
+                    attributeExpressionExecutors[classIndex]).getAttribute().getName();
+            Attribute.Type classLabelAttributeType = inputDefinition.
+                    getAttributeType(attributeName);
+
+            //class label should be String or Bool
+            if (isLabelType(classLabelAttributeType)) {
+                classLabelVariableExecutor = (VariableExpressionExecutor) attributeExpressionExecutors[classIndex];
+            } else {
+                throw new SiddhiAppValidationException(String.format("[label attribute] in %s th index of " +
+                                "classifierUpdate should be either a %s or a %s but found %s",
+                        classIndex, Attribute.Type.BOOL, Attribute
+                                .Type.STRING, classLabelAttributeType));
+            }
+
+        } else {
+            throw new SiddhiAppValidationException((classIndex) + "th parameter is not " +
+                    "an attribute (VariableExpressionExecutor) present in the stream definition. Found a "
+                    + attributeExpressionExecutors[classIndex].getClass().getCanonicalName());
+        }
+        return classLabelVariableExecutor;
+    }
+
+    /**
+     * @param model
+     * @param noOfFeatures
+     * @return
+     */
+    public static boolean isInitialized(AdaptiveHoeffdingTreeModel model, int noOfFeatures) {
+        boolean initialized;
+        if (model.getStreamHeader() != null) {
+            initialized = true;
+            // validate the model
+            if (noOfFeatures != model.getNoOfFeatures()) {
+                throw new SiddhiAppValidationException(String.format("Model [%s] expects %s " +
+                                "features, but the streamingml:updateHoeffdingTree " +
+                                "specifies %s features.", model.getModelName(), model.getNoOfFeatures(),
+                        noOfFeatures));
+            }
+        } else {
+            initialized = false;
+        }
+        return initialized;
+
```

Comment:
remove line breaks.

Reply:
Fixed in [#d65bc3a9237b1178ce3d2d50ac0fab55709e2e0a](https://github.com/wso2-extensions/siddhi-gpl-execution-streamingml/commit/d65bc3a9237b1178ce3d2d50ac0fab55709e2e0a)